cmd/snap-bootstrap/cmd_scan_disk.go

```diff
@@ -115,13 +145,15 @@ func probePartitions(node string) ([]Partition, error) {
 
 	probe.EnablePartitions(true)
 	probe.SetPartitionsFlags(blkid.BLKID_PARTS_ENTRY_DETAILS)
-	probe.EnableSuperblocks(true)
 
 	if err := probe.DoSafeprobe(); err != nil {
 		return nil, err
 	}
 
-	if !isGpt(probe) {
+	// In fallback mode, which is non-UEFI, it may be
+	// MBR disk schema and not gpt, so we still want to
+	// discover filesystem labels.
+	if !fallbackMode && !isGpt(probe) {
```

Comment:
I think that this condition is never happening, as it is returning if a uuid for a booted partition was found (no fallback) and the partition is MBR, but the former can happen only for GPT partitions. So you could remove it.